internal/querynodev2/segments/segment.go

```diff
@@ -199,6 +199,16 @@ func (s *LocalSegment) isValid() bool {
 	return s.ptr != nil
 }
 
+// Provide ONLY the read lock operations,
+// don't make `mut` public to avoid abusing of the mutex
+func (s *LocalSegment) RLock() {
+	s.mut.RLock()
```

Comment:
rename it as closeLock

Reply:
This protects the whole segmen, not for the `ptr` only, keep it?

Reply:
name it to some better name~